modules/exploits/windows/fileformat/winrar_cve_2023_38831.rb

```diff
@@ -59,8 +59,7 @@ def exploit
       input_file = datastore['INPUT_FILE']
       decoy_name = File.basename(input_file)
       decoy_ext = ".#{File.extname(input_file)[1..]}"
-      payload_name = datastore['PAYLOAD_NAME'] || Rex::Text.rand_text_alpha(8) + '.exe'
-
+      payload_name = datastore['PAYLOAD_NAME'] + '.exe'
```

Comment:
I'm not sure if this new logic is semantically equivalent, would you mind taking a deeper look here to confirm 👀

Reply:
Great catch - it's not, will adjust